Get rid of depslist

Minoru opened the pull request:
Fixes #792.

My current dayjob involves quite a few Makefiles, so I learned a lot more about Make and managed to get this working!

This is just a draft for now because apparently it doesn't work in some CI configurations. Testing by real people is most welcome though, as well as reviews :)

dennisschagt reviewed (approved):
Changes look good.
From my limited testing, this seems to work well

Minoru commented:
This is finally ready! :tada: 

I had to specify a minimum supported GNU Make version, please see the commit message in https://github.com/newsboat/newsboat/pull/2095/commits/0684f78c9b3f01258d23153ce70b37b4eb76763c for details.

I don't think this requires any further review, since the bump is the only change since the last one. So I'm merging this right away.

dennisschagt commented:
Great to have this change in. Thanks!
